src/games/strategy/triplea/util/TransportUtils.java

```diff
@@ -198,4 +213,18 @@ public int compare(final Unit o1, final Unit o2) {
     return Optional.empty();
   }
 
+  private static void loadMaxUnits(final Unit transport, final List<Unit> canBeTransported,
```

Comment:
Ultimately a pivot of the code is probably needed, from procedural to OO. IE: perhaps a transport object that could help absorb some of the complexity, in combination with a unit object, with methods like: `unit.canFitOnTransport(transport)`, and should work with a way to sort the list of units by their transport cost.